universal-login-wallet/src/ui/react/Home/Balance.tsx

```diff
@@ -1,21 +1,29 @@
-import React, {useState, useEffect, useContext} from 'react';
-import {useServices} from '../../hooks';
+import React, {useState, useContext} from 'react';
 import {utils} from 'ethers';
+import {TokenDetailsWithBalance, ETHER_NATIVE_TOKEN} from '@universal-login/commons';
+import {useAsyncEffect} from '@universal-login/react';
+import {WalletModalContext} from '../../../core/entities/WalletModalContext';
 import ReceiveIcon from './../../assets/icons/receive.svg';
 import SendIcon from './../../assets/icons/send.svg';
-import {WalletModalContext} from '../../../core/entities/WalletModalContext';
+import {useServices} from '../../hooks';
 
 const Balance = () => {
   const modalService = useContext(WalletModalContext);
-  const [balance, setBalance] = useState(utils.bigNumberify(0));
-  const {balanceService} = useServices();
+  const [ethBalance, setEthBalance] = useState(utils.bigNumberify(0));
+  const {sdk, walletPresenter} = useServices();
+
+  const setBalanceCallback = (balances: TokenDetailsWithBalance[]) => {
+    const ethDetails = balances.find((token: TokenDetailsWithBalance) => token.address === ETHER_NATIVE_TOKEN.address);
+    const balance = ethDetails === undefined ? utils.bigNumberify(0) : ethDetails.balance;
```

Comment:
Extract formater class method:
`new BalanceFormater(TokenDetailsWithBalance[]).getBalanceOf('0x...')`
and more specific:
`balanceFormater.getEtherbalance()`